class.jetpack-xmlrpc-server.php

```diff
@@ -124,6 +126,37 @@ function remote_authorize( $request ) {
 		return $response;
 	}
 
+	function remote_provision( $request ) {
+		$access_token = $request['access_token'];
+		$wpcom_user_id = $request['wpcom_user_id'];
+		$local_username = $request['local_username'];
+		$user = get_user_by( 'login', $local_username );
+
+		if ( ! $user ) {
+			$user = get_user_by( 'email', $local_username );
+		}
+
+		if ( ! $user ) {
+			return $this->error( 'user_not_found', "The user wasn't found - d'oh!" );
+		}
+
+		require_once JETPACK__PLUGIN_DIR . '_inc/class.jetpack-provision.php';
+
+		wp_set_current_user( $user->ID );
+
+		$args = array(
+			'wpcom_user_id' => $wpcom_user_id
+		);
+
+		$result = Jetpack_Provision::partner_provision( $access_token, $args );
+
+		if ( isset( $result->access_token ) && ! empty( $result->access_token ) ) {
+			unset( $result->access_token );
+		}
+
+		return $result;
```

Comment:
This could be an error. In that case, we're not calling `$this->error()` like other methods in this class are doing.

Reply:
This has been handled a few lines above. 👍

Reply:
Fixed :)